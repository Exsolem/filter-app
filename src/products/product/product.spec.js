import React from 'react'
import {shallow} from 'enzyme'

import Product  from './product'


describe('Search', () => {
    const props = {
        product: {
            img: 'product/img/url',
            name: 'product name',
            price: 'product price'
        }
    };

    describe('product initial', () => {
        const product = shallow(<Product {...props}/>);
        it('renders properly', () => {
            expect(product.debug()).toMatchSnapshot()
        });
    });
    describe('elements is rendered and equals to props', () => {
        const product = shallow(<Product {...props}/>);
        describe('product image', () => {
            it('is rendered', () => {
                expect(product.find('ProductImg')).toHaveLength(1);
            });
            it('src is correct', () => {
                expect(product.find('ProductImg').prop('src')).toEqual(props.product.img);
            });
        })
        describe('Product name', () => {
            it('is redered', () => {
                expect(product.find('ProductName')).toHaveLength(1)
            })
            it('is redered', () => {
                expect(product.find('ProductName').text()).toEqual(props.product.name)
            })
        })
        describe('Product price', () => {
            it('is redered', () => {
                expect(product.find('ProductPrice')).toHaveLength(1)
            })
            it('is redered', () => {
                expect(product.find('ProductPrice').text()).toEqual(props.product.price + ' GBP')
            })
        })
    })
});