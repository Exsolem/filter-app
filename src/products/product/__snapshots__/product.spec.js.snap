// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product product initial renders properly 1`] = `
"<Container style={{...}} fluid={false}>
  <ForwardRef noGutters={false}>
    <Col xs={true} sm={true} md={12} lg={true}>
      <ProductLi className=\\"flex-column flex-sm-row flex-md-row\\">
        <Col xs={12} sm={5} md={5}>
          <ProductImg src=\\"product/img/url\\" alt=\\"product\\" />
        </Col>
        <Col xs={12} sm={2} md={2}>
           
          <ProductPrice>
            product price
             GBP
          </ProductPrice>
        </Col>
        <Col xs={12} sm={5} md={5}>
           
          <ProductName>
            product name
          </ProductName>
        </Col>
      </ProductLi>
    </Col>
  </ForwardRef>
</Container>"
`;
